### PR TITLE
Modify Operation to print the stack in stack order

### PR DESCRIPTION
**Pull request by fdawg4l**
Also only print the Begin header when an operation is created, and not
when it's inherited or mutated.

### CODE REVIEW

**Comment by fdawg4l** [comment]
@hickeng these are the changes you asked for to make debugging slightly better.  Ping if I missed anything.

**Comment by fdawg4l** [comment]
@caglar10ur , I had to modify the `Logger` in the trace package to specify a `TextFormatter` that handles control codes to do line wrapping.  Yes, really.  Please take a look and if you have a less crappy alternative, I'm all :ear:.

**Comment by hickeng** [comment]
LGTM

File: pkg/trace/trace.go [hunk 1]
```diff
@@ -16,13 +16,21 @@ package trace
...
 	"time"
 
 	log "github.com/Sirupsen/logrus"
 )
 
-var Logger = log.New()
+var Logger = &log.Logger{
+	Out: os.Stderr,
+	// Apparently this forces terminal handling even when Out isn't a TTY (like
+	// on a VCH, where out is currently a serial dev)
```
Comment (fdawg4l) [inline pkg/trace/trace.go]: It's actually a file.  Will fix this.

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Comment by fdawg4l** [comment]
.

**Pull request merged**